src/lib.rs

```diff
@@ -29,51 +35,89 @@ leptos_i18n::load_locales!();
 pub fn App() -> impl IntoView {
   provide_meta_context();
   provide_i18n_context();
-  provide_query_client();
 
-  let ui_theme = create_rw_signal::<String>(String::from("retro"));
+  let error = create_rw_signal::<Option<LemmyAppError>>(None);
+  provide_context(error);
+  let user = create_rw_signal::<Option<bool>>(None);
+  provide_context(user);
+  let ui_theme = create_rw_signal::<Option<String>>(None);
   provide_context(ui_theme);
 
-  let (is_routing, set_is_routing) = create_signal(false);
+  let ssr_site = create_resource(
+    move || (user.get()),
+    move |_user| async move {
+      let result = LemmyClient.get_site().await;
+
+      match result {
+        Ok(o) => Ok(o),
+        Err(e) => {
+          error.set(Some(e.clone()));
+          Err(e)
+        }
+      }
+    },
+  );
+
+  let site_signal = create_rw_signal::<Option<Result<GetSiteResponse, LemmyAppError>>>(None);
 
   view! {
-    <Router set_is_routing>
+    <Transition fallback=|| {}>
+      // the only way i can find to force a signal to initialize in SSR mode
+      {move || {
+          ssr_site
+              .get()
+              .map(|m| {
+                  site_signal.set(Some(m));
+              });
+      }}
+
+    </Transition>
+
+    <Router>
       <Routes>
-        <Route path="/" view=move || view! { <Layout is_routing/> } ssr=SsrMode::PartiallyBlocked>
-          <Route path="" view=HomeActivity/>
-          <Route path="home" view=PostActivity/>
+        <Route path="/" view=move || view! { <Layout site_signal/> } ssr=SsrMode::Async>
+          <Route path="/*any" view=NotFound/>
 
-          <Route path="communities" view=PostActivity/>
-          <Route path="create_post" view=PostActivity/>
-          <Route path="create_community" view=PostActivity/>
+          <Route path="" view=move || view! { <HomeActivity site_signal/> }/>
 
-          <Route path="search" view=PostActivity/>
-          <Route path="login" view=LoginActivity/>
-          <Route path="signup" view=PostActivity/>
+          <Route path="create_post" view=CommunitiesActivity/>
```

Comment:
Unrelated to this PR, but why did we end up choosing "Activity" as the suffix to our page components?

Reply:
i should have marked the PR as draft again before pushing update - this code may have been removed or changed

Reply:
Unresolving this as it's a broader question about the choice of naming conventions for the page/activity components.